#166 - Added a bounding box query test

omerxx opened the pull request:
#166 

Added a bounding box query test based on our tests framework.
It basically tests 3 aspects:

1. Are the resulted markers in accordance with the given latitude/longitude parameters
2. Whether the markers where correctly filtered given `inaccurate` parameter
3. Whether they were filtered correctly given the severity different parameters

danielhers commented on tests/test_bounding_box_query.py:
```diff
@@ -0,0 +1,41 @@
+
+    def test_location_filters(self):
+        for marker in self.query:
+            self.assertEqual(self.query['sw_lat'] < marker.latitude < self.query['ne_lat'], True)
```
Why not `assertTrue`?

danielhers replied:
Should be `<=`, because that's what the filter asks for.

danielhers commented on tests/__init__.py:
```diff
@@ -0,0 +1 @@
+__author__ = 'Omer'
```
Is this necessary?

omerxx replied:
That allows running a specific commands such as `python -m unittest tests.test_bounding_box_query`
Convenient for testing locally, but I can remove it you don't want it on git

danielhers replied:
Great, keep it then.

danielhers commented:
Nice, it's a good test. However, whenever we run it on an empty database, which is the case in Travis CI automated tests, it will all pass vacuously. Can we make it load some sample data before testing it? Perhaps the 2013 data, since it's the sample data that's on Git? In that case, the year should be 2013 rather than the current year.

Regardless, I'll merge this once you handle the line comments.

omerxx commented:
Not sure what do you mean by making it load the example data.
I changed the default year to 2013 and commented the previous lines for future usability.

Fixed all test calls to `assertTrue` / `assertFalse` where required
Changed to `<=` the bounding box filters

danielhers commented on tests/test_bounding_box_query.py:
```diff
@@ -0,0 +1,45 @@
+
+
+class TestQueryFilters(unittest.TestCase):
+    cyear = str(datetime.datetime.now().strftime("%Y"))
```
You should comment this out too

danielhers commented:
I mean to run `process.py` so that the DB isn't empty when the test is run.
It would mean a lot more time to run the tests, though, so I'm not sure it's worth it. Need to think about it.

omerxx commented:
Done.
